### DISCUSSION TITLE
How can I disabled the signature verification

### DISCUSSION THREAD

**Discussion by ghost** [termux/termux-app#3871] [Q&A] [answered]
I want to install termux in Android Virtual Machine.
But I'm having a problem.
I can't install or update any packages.
APT return a error message:
```
W: GPG error: https://mirrors.tuna.tsinghua.edu.cn/termux/apt/termux-main stable InRelease: At least one invalid signature was encountered.
E: The repository 'https://mirrors.tuna.tsinghua.edu.cn/termux/apt/termux-main stable InRelease' is not signed.
N: Metadata integrity can't be verified, repository is disabled now.
N: Possible causes: unstable or tampered Internet connection, wrong sources.list, outdated keyring or host is down currently.
N: Please note that all hosting problems or other serious issues we announce on our social media pages.
N: Use termux-change-repo for switching to a mirror.
```
It seems like the gpg key signature verification is no pass
I try to use --allow-unauthenticated to disabled signature verification,but it's not working(Maybe my operation was wrong)
The app version is v0.118 and system version is Android 10
My mobile phone is not root because I afraid of the phone will be break down when I   make mistakes(such as rm -rf . /*(. and / has a space))

**Comment by Biswa96** [comment]
Have you tried to change the mirror using `termux-change-repo` command?

**Reply by ghost** [reply]
Yes , I used it to change into others repo
But it will be auto change to official repo

**Comment by sylirre** [comment, marked as answer]
In your case verification fails due to silent crash of GnuPG verification utility. This is not a bug of Termux but rather issue of your runtime environment.

Noting that applications **VMOS**, **F1VM** and their equivalents are not supported by Termux. It is not possible for us to provide support for non-standard environments, especially when they are proprietary.

These apps are not virtual machines and run a highly modified version of Android OS under proot-like software container. This implies possibility of issues which can't happen under normal circumstances.

Verification failure is not only issue which happens on VMOS/F1VM.
